pkg/util/steps/action.go

```diff
@@ -17,17 +17,31 @@ type actionFunction func(context.Context) error
 
 // Action returns a Step which will execute the action function `f`. Errors from
 // `f` are returned directly.
-func Action(f actionFunction) Step {
-	return actionStep{f}
+func Action(f actionFunction, metricsTopic ...string) Step {
+	return actionStep{
+		f:            f,
+		metricsTopic: metricsTopic,
+	}
 }
 
+// name field is for better naming
+// when processing metrics emitting
 type actionStep struct {
-	f actionFunction
+	f            actionFunction
+	metricsTopic []string
```

Comment:
I'm not following why we have an array of string when we only consider the first element in it and we only ever pass one string into the `Action` function. Can this be changed to just accept a `metricsTopic string`? The same applies for `Condition` and `AuthorizationRefreshingAction`

Reply:
@s-amann Hi Spencer, thanks for the feedback, the reason why I am using a slice here is to make the param `metricsTopic` optional, so we do not need to pass this param if it is not needed in some other places. If we make it as a `string` field, then we have to pass an empty string when we dun want to set any metrics topic for the certain actions.

Reply:
Ah, I see. Perhaps we should modify the comment above the functions to include `the metricsTopic parameter is optional and only the first element is considered if present` so the comment would eventually look like:
```
// Action returns a Step which will execute the action function `f`.
// The metricsTopic parameter is optional and only the first element is considered if present.
// Errors from `f` are returned directly.
func Action(f actionFunction, metricsTopic ...string) Step {
```
What do you think @ArrisLee ?